pkg/quota/generic/evaluator.go

```diff
@@ -41,6 +41,13 @@ func ListResourceUsingInformerFunc(f informers.SharedInformerFactory, resource s
 		if err != nil {
 			return nil, err
 		}
+
+		// when the informer has not synced, should return error
+		// otherwise, the quota will be exceeded.
+		if !informer.Informer().HasSynced() {
+			return nil, fmt.Errorf("%s hasn't been synced", resource.String())
```

Comment:
i have checked in the resourceQuota controller that it has already wait until synced, see [this](https://github.com/kubernetes/kubernetes/blob/v1.7.0/pkg/controller/resourcequota/resource_quota_controller.go#L131-#L144) and [this](https://github.com/kubernetes/kubernetes/blob/v1.7.0/pkg/controller/resourcequota/resource_quota_controller.go#L249), so this fix is no need. :)